lib/validate-config.js

```diff
@@ -56,6 +56,51 @@ function validateConfig(config, configPath) {
 
   const configWithDefaults = Object.assign({}, defaults, config);
 
+  const validConfigProperties = [
+    'siteBasePath',
+    'siteOrigin',
+    'wrapperPath',
+    'notFoundPath',
+    'externalStylesheets',
+    'autoprefixerBrowsers',
+    'pagesDirectory',
+    'staticDirectory',
+    'outputDirectory',
+    'temporaryDirectory',
+    'data',
+    'dataSelectors',
+    'vendorModules',
+    'webpackLoaders',
+    'webpackPlugins',
+    'babelPlugins',
+    'babelExclude',
+    'postcssPlugins',
+    'fileLoaderExtensions',
+    'jsxtremeMarkdownOptions',
+    'inlineJs',
+    'production',
+    'port',
+    'verbose'
+  ];
+
+  const configWithDefaultsProperties = Object.keys(configWithDefaults);
+
+  const invalidProperties = _.pullAll(
+    configWithDefaultsProperties,
+    validConfigProperties
+  );
```

Comment:
Nice method for getting the full list of invalid properties, instead of just error on the first one!